core/src/HIP/Kokkos_HIP_ParallelFor_Range.hpp

```diff
@@ -63,8 +63,8 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::HIP> {
 
     using DriverType = ParallelFor<FunctorType, Policy, Kokkos::HIP>;
     const int block_size =
-        Kokkos::Impl::hip_get_preferred_blocksize<DriverType, LaunchBounds>(
-            m_policy.space().hip_device());
+        Kokkos::Impl::get_preferred_blocksize_range<DriverType, LaunchBounds>(
```

Comment:
Since the function is only used for `parallel_for` maybe rename it to sth like `get_preferred_blocksize_for_range`